tests/repo_finder/test_repo_finder.py

```diff
@@ -69,3 +75,38 @@ def test_resolve_analysis_target(
 ) -> None:
     """Test the resolve analysis target method with valid inputs."""
     assert Analyzer.to_analysis_target(config, available_domains) == expect
+
+
+def test_get_commit_from_version() -> None:
```

Comment:
I think the thing you are actually testing in this test method is the mapping from PURLs (or more specifically the version part in PURLs) to tags.
In that case, I think mocking a Git repo is unnecessary. I think we can simplify matters here by not unit-testing directly on `get_commit_from_version(git_obj: Git, purl: PackageURL) ->  [Branch, Commit]`. Instead, we can create a smaller function that `get_commit_from_version` itself calls within, but have a simpler interface, something along the line of:

```
find_matching_tags(tags: Tags, purl: PackageURL) -> [Branch, Commit]
```

and unit test this instead, which is much easier to handle without any complex mocking setup.

Reply:
I will look into this as part of adding tests with Hypothesis.